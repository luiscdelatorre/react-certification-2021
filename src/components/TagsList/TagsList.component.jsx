import React from 'react';
import { useHistory } from 'react-router';
import { useSearch } from '../../providers/Search';
import Styled from './TagsList.styles';

const TagsList = ({ tags }) => {
  const { search } = useSearch();
  const history = useHistory();

  const searchTag = (query) => {
    search(query);
    history.push('/');
  };

  return (
    <Styled.TagList>
      {tags
        .filter((tag) => tag.length > 2 && tag.length < 20)
        .map((tag) => {
          return (
            <Styled.TagListItem key={`${tag}`}>
              <Styled.Tag
                type="button"
                onClick={() => searchTag(tag)}
                data-testid="tag-button"
              >
                {tag}
              </Styled.Tag>
            </Styled.TagListItem>
          );
        })}
    </Styled.TagList>
  );
};

export default TagsList;
