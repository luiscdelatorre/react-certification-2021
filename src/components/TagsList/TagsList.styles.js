import styled from 'styled-components';

const TagList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  margin-top: 1rem;
`;

const TagListItem = styled.li`
  display: inline-block;
  padding: 0 1rem 1rem 0;
`;

const Tag = styled.button`
  background-color: ${(props) => props.theme.tagSecondaryBackground};
  color: ${(props) => props.theme.tagSecondaryText};
  display: inline-block;
  padding: 0.5rem 1rem;
  border-radius: ${(props) => props.theme.borderRadiusSm};
  letter-spacing: 0;
  margin: 0;
  text-transform: capitalize;
  line-height: 1.4rem;
`;

const Styled = { TagList, TagListItem, Tag };
export default Styled;
