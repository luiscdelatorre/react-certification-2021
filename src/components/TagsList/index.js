export { default } from './TagsList.component';
