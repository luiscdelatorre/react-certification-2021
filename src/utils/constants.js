const AUTH_STORAGE_KEY = 'wa_cert_authenticated';
const SEARCH_STORAGE_KEY = 'wa_cert_search';
const API_URL = `https://www.googleapis.com/youtube/v3`;

export { AUTH_STORAGE_KEY, SEARCH_STORAGE_KEY, API_URL };
