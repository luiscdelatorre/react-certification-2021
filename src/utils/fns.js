const random = (limit) => {
  return Math.floor(Math.random() * limit);
};

const formatDate = (dateString) => {
  const date = new Date(dateString);
  return date.toLocaleDateString();
};

export { random, formatDate };
