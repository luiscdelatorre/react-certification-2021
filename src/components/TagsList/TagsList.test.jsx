import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import TagsList from './TagsList.component';
import SearchProvider from '../../providers/Search';
import { storage } from '../../utils/storage';
import { SEARCH_STORAGE_KEY } from '../../utils/constants';

const tags = ['Foo', 'Bar', 'Baz'];

describe('Tags List component', () => {
  it('Should display mock data in component', () => {
    render(
      <BrowserRouter>
        <SearchProvider>
          <TagsList tags={tags} />
        </SearchProvider>
      </BrowserRouter>
    );

    const totalMenuItems = screen.getAllByRole('listitem').length;
    expect(totalMenuItems).toBe(3);
  });

  it('trigger a new search from selected tag', () => {
    render(
      <BrowserRouter>
        <SearchProvider>
          <TagsList tags={tags} />
        </SearchProvider>
      </BrowserRouter>
    );

    const button = screen.queryAllByTestId('tag-button')[0];
    fireEvent.click(button);
    expect(storage.get(SEARCH_STORAGE_KEY)).toBe(tags[0]);
  });

  it('Validates TagsList snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <SearchProvider>
          <TagsList tags={tags} />
        </SearchProvider>
      </BrowserRouter>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
