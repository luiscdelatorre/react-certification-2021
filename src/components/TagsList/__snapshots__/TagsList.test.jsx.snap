// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags List component Validates TagsList snapshot 1`] = `
<ul
  className="sc-bdfBwQ gcWJNd"
>
  <li
    className="sc-gsTCUz kZitFI"
  >
    <button
      className="sc-dlfnbm bPibKy"
      data-testid="tag-button"
      onClick={[Function]}
      type="button"
    >
      Foo
    </button>
  </li>
  <li
    className="sc-gsTCUz kZitFI"
  >
    <button
      className="sc-dlfnbm bPibKy"
      data-testid="tag-button"
      onClick={[Function]}
      type="button"
    >
      Bar
    </button>
  </li>
  <li
    className="sc-gsTCUz kZitFI"
  >
    <button
      className="sc-dlfnbm bPibKy"
      data-testid="tag-button"
      onClick={[Function]}
      type="button"
    >
      Baz
    </button>
  </li>
</ul>
`;
